osquery/CMakeLists.txt

```diff
@@ -34,6 +34,11 @@ endif()
 
 # Add default linking details (the first argument means SDK + core).
 ADD_OSQUERY_LINK(TRUE "-rdynamic")
+foreach(DISTRO ${OSQUERY_REQUIRE_RUNTIMES})
+  if(${OSQUERY_BUILD_DISTRO} STREQUAL ${DISTRO})
+    ADD_OSQUERY_LINK(FALSE "-static-libstdc++")
```

Comment:
This is a much debated path to support the newer versions of RocksDB and allow us flexibility in using more C++11 features. Unless we find problems in practice I'd rather include it statically for Rocks and our code than ship a DSO and an RPATH. This means less moving parts at the cost of some ctor/dtor or other related static-inclusion confound.